crates/services/gas_price_service/src/fuel_gas_price_updater/tests.rs

```diff
@@ -38,15 +38,13 @@ impl FakeMetadata {
 }
 
 impl MetadataStorage for FakeMetadata {
-    fn get_metadata(
-        &self,
-        _block_height: &BlockHeight,
-    ) -> Result<Option<UpdaterMetadata>> {
-        Ok(self.inner.lock().unwrap().clone())
+    fn get_metadata(&self, _: &BlockHeight) -> Result<Option<UpdaterMetadata>> {
```

Comment:
maybe not related to this MR, but is there any reason why FakeMetadata::inner is wrapped around an `Arc`? Mutex implements `Send and Sync` already, and

Reply:
@mitchturner perhaps?

Reply:
So it's clone-able and we can access a copy of the inner value outside of the task it's sent to.

Quickly looking through the code I don't see anyplace we are using that functionality, but I think we probably should be in the `next__new_l2_block_saves_old_metadata` test.